doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,851 @@
+    /// Defines the trust level of the app catalog.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum AppCatalogTrustLevel
+    {
+        None,
+        Trusted,
+    };
+
+    /// IMPLEMENTATION NOTE: SourceDetails from AppInstallerRepositorySource.h
+    /// Interface for retrieving information about an app catalog without acting on it.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass AppCatalogInfo
+    {
+        /// The app catalog's unique identifier.
+        String Id { get; };
+        /// The name of the app catalog.
+        String Name { get; };
+        /// The type of the app catalog.
+        String Type { get; };
+        /// The argument used when adding the app catalog.
+        String Arg { get; };
+        /// The app catalog's extra data string.
+        String ExtraData { get; };
+        /// The last time that this app catalog was updated.
+        Windows.Foundation.DateTime LastUpdateTime { get; };
+        /// The origin of the app catalog.
+        AppCatalogOrigin Origin { get; };
+        /// The trust level of the app catalog
+        AppCatalogTrustLevel TrustLevel { get; };
+    };
+
+    /// A metadata item of a package version.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum PackageVersionMetadata
+    {
+        /// The InstallerType of an installed package
+        InstalledType,
+        /// The Scope of an installed package
+        InstalledScope,
+        /// The system path where the package is installed
+        InstalledLocation,
+        /// The standard uninstall command; which may be interactive
+        StandardUninstallCommand,
+        /// An uninstall command that should be non-interactive
+        SilentUninstallCommand,
+        /// The publisher of the package
+        Publisher,
+        /// The locale of the package
+        Locale,
+    };
+
+    /// IMPLEMENTATION NOTE: IPackageVersion from AppInstallerRepositorySearch.h
+    /// A single package version.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass PackageVersionInfo
+    {
+        /// IMPLEMENTATION NOTE: PackageVersionMetadata fields from AppInstallerRepositorySearch.h
+        /// DESIGN NOTE: These would be a good candidate to leave out in V1. Maybe just keep InstalledLocation
+        /// and InstalledScope?
+        /// Gets any metadata associated with this package version.
+        /// Primarily stores data on installed packages.
+        /// Metadata fields may have no value (e.g. packages that aren't installed will not have an InstalledLocation).
+        String GetMetadata(PackageVersionMetadata metadataType);
+        /// IMPLEMENTATION NOTE: PackageVersionProperty fields from AppInstallerRepositorySearch.h
+        String Id { get; };
+        String Name { get; };
+        String AppCatalogIdentifier { get; };

```

Comment:
Are AppCatalogIdentifier and AppCatalogName necessary if you have the AppCatalog itself a few lines down?

Reply:
No. Good point.